src/Main.ts

```diff
@@ -810,7 +811,20 @@ export class Main {
         }
         const port = this.config.homeserver.appservice_port || cliPort;
         this.bridge.run(port, this.config, this.appservice);
-        const roomListPromise = this.bridge.getBot().getJoinedRooms() as Promise<string[]>;
+        let joinedRooms: string[]|null = null;
+        while(joinedRooms === null) {
```

Comment:
I wish there was an event handler for this. While I guess the async function || timeout are achieving the same, this looks scary. (thinking never-ending loop or 100% CPU core usage).

The more I read through this, the less it concerns me though.. looks like it's safe.

Reply:
I believe this is safe. The CPU won't spin while waiting for the await to complete, so there is no harm.